model/quic-socket-base.cc

```diff
@@ -1581,8 +1612,15 @@ QuicSocketBase::Close (void)
       and m_socketState != CLOSING)   //Connection Close from application signal
     {
       SetState (CLOSING);
-      m_drainingPeriodEvent.Cancel ();
-      m_idleTimeoutEvent.Cancel ();
+      if (m_flushOnClose)
```

Comment:
Check here - are you sending two connectionclose packets?

Reply:
Whoops, I'll update